VisionTextDualEncoder (#13511)
huggingface/transformers: docs/source/model_doc/vision_text_dual_encoder.rst

.. 

>     Copyright 2020 The HuggingFace Team. All rights reserved.

Author: ```suggestion
    Copyright 2021 The HuggingFace Team. All rights reserved.
```

>     Copyright 2020 The HuggingFace Team. All rights reserved.

Reviewer: ```suggestion
    Copyright 2021 The HuggingFace Team. All rights reserved.
```

    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
    the License. You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
    specific language governing permissions and limitations under the License.

VisionTextDualEncoder
-----------------------------------------------------------------------------------------------------------------------

Overview
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The :class:`transformers.VisionTextDualEncoderModel` can be used to initialize a vision-text dual encoder model with

Reviewer 2: ```suggestion
The :class:`~transformers.VisionTextDualEncoderModel` can be used to initialize a vision-text dual encoder model with
```

any pretrained vision autoencoding model as the vision encoder (*e.g.* :doc:`ViT <vit>`, :doc:`BEiT <beit>`, :doc:`DeiT

> <deit>`) and any pretrained text model as the text encoder (*e.g.* :doc:`RoBERTa <roberta>`, :doc:`BERT <bert>`). This

Reviewer 2: Not really *any* text model, it can't be GPT-2 for instance. I think the difference with the `VisionEncoderDecoder` architectures should be made clearer.

> model adds two projection layers to project the vision and text embeddings to an embedding space with identical

Reviewer: ```suggestion
Two projection layers are added on top of both the vision and text encoder to project the output embeddings to a shared latent space.
```

might be a bit cleaner

may

dimentions. The projection layers are randomly initialized so the model should be fine-tuned on a downstream task. This
model can be used to align the vision-text embeddings using CLIP like contrastive image-text training and then can be
used for zero-shot vision tasks such image-classification or retrieval.

In `LiT: Zero-Shot Transfer with Locked-image Text Tuning <https://arxiv.org/abs/2111.07991>`__ it is shown how

> leveraging pre-trained (locked/frozen) image and text model yields for contrastive learning yields significant

Reviewer: ```suggestion
leveraging pre-trained (locked/frozen) image and text model yields for contrastive learning significant
```

> improvment on new zero-shot vision tasks such as image classification or retrieval.

Reviewer: ```suggestion
improvement on zero-shot image classification or retrieval.
```
